### PR TITLE
Add ApplicationProjects handlers and repository

### PR DESCRIPTION
**Pull request by dawid-pawulski**

### CODE REVIEW

File: src/HE.InvestmentLoans.BusinessLogic/LoanApplication/ApplicationProject/CommandHandlers/DeleteProjectCommand.cs [hunk 1]
```diff
@@ -0,0 +1,6 @@
+using HE.InvestmentLoans.BusinessLogic.LoanApplication.Entities;
+using MediatR;
+
+namespace HE.InvestmentLoans.BusinessLogic.LoanApplication.ApplicationProject.CommandHandlers;
+
+public record DeleteProjectCommand(LoanApplicationEntity LoanApplicationEntity, Guid ProjectId) : IRequest;
```
Comment (lukasz-ciura-pwc) [inline src/HE.InvestmentLoans.BusinessLogic/LoanApplication/ApplicationProject/CommandHandlers/DeleteProjectCommand.cs:6]: `LoanApplicationEntity` shouldn't be needed

File: src/HE.InvestmentLoans.BusinessLogic/LoanApplication/ApplicationProject/Entities/ApplicationProjects.cs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        var project = new Project
+        {
+            Name = GenerateNextProjectName(),
+        };
+
+        Projects.Add(project);
+
+        return project;
+    }
+
```
Comment (lukasz-ciura-pwc) [inline src/HE.InvestmentLoans.BusinessLogic/LoanApplication/ApplicationProject/Entities/ApplicationProjects.cs:29]: DeleteProject should be part of this aggregate.

File: src/HE.InvestmentLoans.BusinessLogic/LoanApplication/ApplicationProject/Entities/Project.cs [hunk 3]
```diff
@@ -0,0 +1,252 @@
+using HE.InvestmentLoans.BusinessLogic.LoanApplicationLegacy.Workflow;
+using HE.InvestmentLoans.Common.Utils;
+
+namespace HE.InvestmentLoans.BusinessLogic.LoanApplication.ApplicationProject.Entities;
+
+public class Project
```
Comment (lukasz-ciura-pwc) [inline src/HE.InvestmentLoans.BusinessLogic/LoanApplication/ApplicationProject/Entities/Project.cs:6]: We need to talk about it ^_^

File: src/HE.InvestmentLoans.BusinessLogic/LoanApplication/ApplicationProject/QueryHandlers/GetAllApplicationProjectsQueryHandler.cs [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    {
+        _applicationProjectsRepository = applicationProjectsRepository;
+        _loanUserContext = loanUserContext;
+    }
+
+    public async Task<ApplicationProjects> Handle(GetAllApplicationProjectsQuery request, CancellationToken cancellationToken)
+    {
+        return _applicationProjectsRepository.GetAllProjects(
+                                                request.LoanApplicationId,
+                                                new UserAccount(_loanUserContext.UserGlobalId, await _loanUserContext.GetSelectedAccountId().ConfigureAwait(false)));
```
Comment (lukasz-ciura-pwc) [inline src/HE.InvestmentLoans.BusinessLogic/LoanApplication/ApplicationProject/QueryHandlers/GetAllApplicationProjectsQueryHandler.cs:23]: `.ConfigureAwait(false)` is it needed?

File: src/HE.InvestmentLoans.BusinessLogic/LoanApplication/ApplicationProject/QueryHandlers/GetProjectDetailsQuery.cs [hunk 5]
```diff
@@ -0,0 +1,7 @@
+using HE.InvestmentLoans.BusinessLogic.LoanApplication.ApplicationProject.Entities;
+using HE.InvestmentLoans.Contract.Application.ValueObjects;
+using MediatR;
+
+namespace HE.InvestmentLoans.BusinessLogic.LoanApplication.ApplicationProject.QueryHandlers;
+
+public record GetProjectDetailsQuery(LoanApplicationId LoanApplicationId, Guid ProjectId) : IRequest<Project>;
```
Comment (lukasz-ciura-pwc) [inline src/HE.InvestmentLoans.BusinessLogic/LoanApplication/ApplicationProject/QueryHandlers/GetProjectDetailsQuery.cs:7]: To discuss: should Query Handles can return Entities?

File: src/HE.InvestmentLoans.BusinessLogic/LoanApplication/ApplicationProject/Repositories/ApplicationProjectsRepository.cs [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+    {
+        var loanApplicationSessionModel = _httpContextAccessor.HttpContext?.Session.Get<LoanApplicationViewModel>(loanApplicationEntity.Id.ToString());
+        var projectToUpdate = loanApplicationSessionModel.Projects.FirstOrDefault(p => p.Id == project.Id);
+
+        if (projectToUpdate == null)
+        {
+            // throw some error
+        }
+
+        loanApplicationSessionModel.Timestamp = DateTime.Now;
```
Comment (lukasz-ciura-pwc) [inline src/HE.InvestmentLoans.BusinessLogic/LoanApplication/ApplicationProject/Repositories/ApplicationProjectsRepository.cs:46]: Use `DateTimeProvider`. But I think we don't have to set it, CRM will do it for us.

**Review by lukasz-ciura-pwc — CHANGES_REQUESTED** [review]

**Review by lukasz-ciura-pwc — APPROVED** [review]

**Review by jakub-lotko — APPROVED** [review]

**Pull request merged**